### ISSUE TITLE
vkCreateDevice behavior depends on order of pQueueCreateInfos

### ISSUE THREAD

**Issue by samcjohnston** [KhronosGroup/MoltenVK#697]
I've noticed that creating a device where the VkDeviceQueueCreateInfo elements are not ordered by ascending queueFamilyIndex can cause vkGetDeviceQueue to either return invalid queues or crash. Below is a minimal repro that crashes on my machine:

```c++
VkDeviceQueueCreateInfo queueCreateInfo[3] = {};
queueCreateInfo[0].sType = VK_STRUCTURE_TYPE_DEVICE_QUEUE_CREATE_INFO;
queueCreateInfo[0].queueFamilyIndex = 1; // NOTE: starting with 1
queueCreateInfo[0].queueCount = 1;
queueCreateInfo[0].pQueuePriorities = &queuePriority;
    
queueCreateInfo[1].sType = VK_STRUCTURE_TYPE_DEVICE_QUEUE_CREATE_INFO;
queueCreateInfo[1].queueFamilyIndex = 0;  // NOW down to 0
queueCreateInfo[1].queueCount = 1;
queueCreateInfo[1].pQueuePriorities = &queuePriority;
    
queueCreateInfo[2].sType = VK_STRUCTURE_TYPE_DEVICE_QUEUE_CREATE_INFO;
queueCreateInfo[2].queueFamilyIndex = 2;
queueCreateInfo[2].queueCount = 1;
queueCreateInfo[2].pQueuePriorities = &queuePriority;
    
VkDeviceCreateInfo createInfo = {};
createInfo.sType = VK_STRUCTURE_TYPE_DEVICE_CREATE_INFO;
createInfo.pQueueCreateInfos = queueCreateInfo;
createInfo.queueCreateInfoCount = queueInfoCount;
createInfo.pEnabledFeatures = &deviceFeatures;
/* etc */
    
VkResult result = vkCreateDevice(physicalDevice, &createInfo, nullptr, &device);
    
VkQueue queue;
vkGetDeviceQueue(device, 1, 0, &queue); // WILL CRASH!!
```

I believe this is due to the resize inside MVKDevice::initQueues. When run on the pQueueCreateInfos from the example above, the first iteration will resize the _queuesByQueueFamilyIndex to 2 elements to accommodate queue family index 1, then the second iteration will resize _queuesByQueueFamilyIndex to 1 element to accommodate queue family index 0, thus destroying the element at index 1 that was created on the first iteration.

Obviously, can be worked around by providing queues in sorted order!

**Comment by billhollings** [comment]
Fixed in PR #699. Please retest and close if fixed.

**Comment by samcjohnston** [comment]
My test case no longer crashes. Looks good! Thank you.
